.github/workflows/promote.yml

```diff
@@ -0,0 +1,48 @@
+name: Promote
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Release version (e.g. v0.1.0)'
+        required: true
+      channel:
+        description: 'Release channel'
+        required: true
+        default: 'stable'
+
+env:
+  # Common users. We can't run a step 'if secrets.AWS_USR != ""' but we can run
+  # a step 'if env.AWS_USR' != ""', so we copy these to succinctly test whether
+  # credentials have been provided before trying to run steps that need them.
+  DOCKER_USR: ${{ secrets.DOCKER_USR }}
+  AWS_USR: ${{ secrets.AWS_USR }}
```

Comment:
```suggestion
```
I don't think we need AWS user for providers since they don't publish an artifact other than a Docker image.

Reply:
You're probably right, but I kind of want to keep this as is. Our general pattern is to pretty much try keep the CI file the same everywhere so we can just copy and paste to propagate new versions with minimal updates.